src/app-layer-parser.c

```diff
@@ -597,12 +597,12 @@ uint64_t AppLayerParserGetTransactionLogId(AppLayerParserState *pstate)
     SCReturnCT((pstate == NULL) ? 0 : pstate->log_id, "uint64_t");
 }
 
-void AppLayerParserSetTransactionLogId(AppLayerParserState *pstate)
+void AppLayerParserSetTransactionLogId(AppLayerParserState *pstate, uint64_t tx_id)
 {
     SCEnter();
 
     if (pstate != NULL)
-        pstate->log_id++;
+        pstate->log_id = tx_id;
```

Comment:
That could happen if tx1 was free'd and removed fromthe list before tx0 because tx1 is done and tx0 is not.  I'm not sure if that can even happen right now.  But yeah, its a simple check and we don't want it going backwards.